olive/passes/onnx/optimum_conversion.py

```diff
@@ -48,18 +48,30 @@ def _default_config(cls, accelerator_spec: AcceleratorSpec) -> Dict[str, PassCon
                 default_value=None,
                 description="Extra arguments to pass to the `optimum.exporters.onnx.main_export` function.",
             ),
+            "custom_model": PassConfigParam(
```

Comment:
>custom_model [](http://example.com/codeflow?start=13&length=12)

should we use another pass for the conversion? It seems like a different conversion strategy.

Reply:
It seems has same args and similar func with [main_export](https://github.com/huggingface/optimum/blob/d03ab100206cb9f0e62167a36ee6997424bb9bb5/optimum/exporters/onnx/__main__.py#L47) but [onnx_export_from_model](https://github.com/huggingface/optimum/blob/d03ab100206cb9f0e62167a36ee6997424bb9bb5/optimum/exporters/onnx/convert.py#L901) accept a modified loaded model.

Reply:
yes, that's the reason I say they are different loading strategy. @jambayk, do you think if it need to be in another separated pass instead of by using a hacky flag?

Reply:
Like I mentioned in this comment https://github.com/microsoft/Olive/pull/957#discussion_r1522208907, if the gptq pass is limited to hf models, then I think the previous method of quantizing and exporting the model in the same pass is good enough? 

There is not much to be gained if we cannot save a quantized custom model class.

Reply:
If want the keep the workflow of gptq quantized pytorch model -> export, I don't have strong opinions about using a flag in this pass or creating a new pass. Looks about the same to me.

Reply:
I think it should be not limited to hf modelsI made it to two steps only to meet a 1p customer's need, they have custom model and custom export and need gptq quantization. Is there better way to do that? Can I import the conversion code from OnnxConversion to do the custom export?

Reply:
If the `torch.save` method works for custom model, I think it is fine to make it two steps. It's just that in my local testing, I could not save a custom model I defined in a user script. 

For the optimum export question Mike brought up, I don't have opinions about a new optimum part is needed or not.

Reply:
If the custom model save doesn't work, I think the gptq quantization pass could maybe provide an option to use optimum exporter or torch.onnx exporter. The export code in the conversion pass is a static method so it can be called directly if needed https://github.com/microsoft/Olive/blob/2eb9f384db86bc0f1cf33b0b2c9cdbc2602d7078/olive/passes/onnx/conversion.py#L145. 

This does mean duplicate work for export to ideally the two step workflow makes more sense.